tests/patmat/t8511.check

```diff
@@ -1 +1 @@
-18: Pattern Match Exhaustivity: Baz(), Bar(_)
+18: Pattern Match Exhaustivity: EatsExhaustiveWarning(_), Baz(), Bar(_)
```

Comment:
I'm not too sure why no warning was emitted for `EatsExhaustiveWarning` here. In the test [code](https://github.com/AleksanderBG/dotty/blob/b66def8805200ae2e01e0a436b331613cb37880a/tests/patmat/t8511.scala), `Reference` is pattern-matched with a possibly-failing `unapply` method (it returns `Option` and not `Some`), so it seems that a warning _should_ be emitted. @liufengyun - do you recall why the original code behaves as it does?

Reply:
This is because there's no known children of `Reference`, thus it's exhaustive.